src/module_wrap.cc

```diff
@@ -455,15 +460,36 @@ static Local<Object> createImportAttributesContainer(
   return attributes;
 }
 
+static std::vector<std::pair<std::string, std::string>>
+createImportAttributesPairs(Local<Context> context,
+                            Isolate* isolate,
+                            Local<FixedArray> raw_attributes,
+                            const int elements_per_attribute) {
+  CHECK_EQ(raw_attributes->Length() % elements_per_attribute, 0);
+  size_t num_attributes = raw_attributes->Length() / elements_per_attribute;
+  std::vector<std::pair<std::string, std::string>> attributes;
+  attributes.reserve(num_attributes);
+
+  for (int i = 0; i < raw_attributes->Length(); i += elements_per_attribute) {
+    Utf8Value key_utf8(isolate, raw_attributes->Get(context, i).As<String>());
+    Utf8Value value_utf8(isolate,
+                         raw_attributes->Get(context, i + 1).As<String>());
+
+    attributes.emplace_back(key_utf8.ToString(), value_utf8.ToString());
```

Comment:
Formally speaking, we should probably either use a commutative hashing function or ensure that this list is ordered consistently so that import attribute ordering does not affect identity.